src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncReceiptsRequestHandler.scala

```diff
@@ -1,131 +0,0 @@
-import akka.util.ByteString
-import io.iohk.ethereum.blockchain.sync.FastSyncReceiptsRequestHandler.ReceiptsValidationResult
-import io.iohk.ethereum.blockchain.sync.FastSyncReceiptsRequestHandler.ReceiptsValidationResult._
-import io.iohk.ethereum.db.storage.AppStateStorage
-import io.iohk.ethereum.domain.{Blockchain, Receipt}
-import io.iohk.ethereum.network.Peer
-import io.iohk.ethereum.network.p2p.messages.PV63.{GetReceipts, Receipts}
-import io.iohk.ethereum.validators.BlockValidator
-import io.iohk.ethereum.validators.BlockValidator.BlockError
-import org.spongycastle.util.encoders.Hex
-
-import scala.concurrent.duration.FiniteDuration
-
-class FastSyncReceiptsRequestHandler(
-    peer: Peer,
-    peerResponseTimeout: FiniteDuration,
-    etcPeerManager: ActorRef,
-    peerMessageBus: ActorRef,
-    requestedHashes: Seq[ByteString],
-    appStateStorage: AppStateStorage,
-    blockchain: Blockchain,
-    blockValidator: BlockValidator)(implicit scheduler: Scheduler)
-  extends SyncRequestHandler[GetReceipts, Receipts](peer, peerResponseTimeout, etcPeerManager, peerMessageBus) {
-
-  override val requestMsg = GetReceipts(requestedHashes)
-  override val responseMsgCode: Int = Receipts.code
-
-  override def handleResponseMsg(receipts: Receipts): Unit = {
-    validateReceipts(requestedHashes, receipts) match {
-      case ReceiptsValid(blockHashesWithReceipts) =>
-        blockHashesWithReceipts.foreach { case (hash, receiptsForBlock) =>
-          blockchain.save(hash, receiptsForBlock)
-        }
-
-        val receivedHashes = blockHashesWithReceipts.unzip._1
-        updateBestBlockIfNeeded(receivedHashes)
-
-        if (receipts.receiptsForBlocks.isEmpty) {
-          val reason = s"got empty receipts for known hashes: ${requestedHashes.map(h => Hex.toHexString(h.toArray[Byte]))}"
-          syncController ! BlacklistSupport.BlacklistPeer(peer.id, reason)
-        }
-
-        val remainingReceipts = requestedHashes.drop(receipts.receiptsForBlocks.size)
-        if (remainingReceipts.nonEmpty) {
-          syncController ! FastSync.EnqueueReceipts(remainingReceipts)
-        }
-
-        log.info("Received {} receipts in {} ms", receipts.receiptsForBlocks.size, timeTakenSoFar())
-
-      case ReceiptsInvalid(error) =>
-        val reason =
-          s"got invalid receipts for known hashes: ${requestedHashes.map(h => Hex.toHexString(h.toArray[Byte]))}" +
-          s" due to: $error"
-        syncController ! BlacklistSupport.BlacklistPeer(peer.id, reason)
-        syncController ! FastSync.EnqueueReceipts(requestedHashes)
-
-      case ReceiptsDbError =>
-        syncController ! FastSync.RedownloadBlockchain
-    }
-    cleanupAndStop()
-  }
-
-  private def updateBestBlockIfNeeded(receivedHashes: Seq[ByteString]): Unit = {
-    val fullBlocks = receivedHashes.flatMap { hash =>
-      for {
-        header <- blockchain.getBlockHeaderByHash(hash)
-        _ <- blockchain.getBlockBodyByHash(hash)
```

Comment:
Should we incorporate this check for having the block bodies to the [`updateBestBlockIfNeeded` implementation in Fast Sync](https://github.com/input-output-hk/mantis/blob/f443f690ab36de73507a98d32195ec0186c20d9f/src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala#L549)?